### PR TITLE
Implement elm-analyse like diagnostics natively using tree-sitter

### PR DESCRIPTION
**Pull request by jmbockhorst**
Had some fun using tree-sitter queries to start implementing some elm-analyse functionality. Currently, there is diagnostics for unsused import value and type, and unused import alias. Still working on more, just wanted to show the progress and how powerful using queries are. 

### Elm Analyse checks implemented:
- [x] Boolean Case Expression
- [x] Drop Concat Of Lists
- [x] Drop Cons Of Item And List
- [x] Fully Applied Operator as Prefix
- [x] Map Nothing To Nothing
- [x] No Top Level Signature
- [x] Single Field Record
- [x] Unnecessary List Concat
- [x] Unnecessary Port Module
- [x] Unused Import
- [x] Unused Import Alias
- [x] Unused Imported Variable
- [x] Unused Pattern Variable
- [x] Unused Top Level
- [x] Unused Type Alias
- [x] Unused Value Constructor
- [x] Unused Variable
- [x] Use Cons Over Concat

### CODE REVIEW

File: src/util/refactorEditUtils.ts [hunk 1]
```diff
@@ -50,22 +50,29 @@ export class RefactorEditUtils {
       const exposedValues = TreeUtils.descendantsOfType(
```
Comment (Razzeee) [inline src/util/refactorEditUtils.ts:50]: Might want to rename this

**Comment by jmbockhorst** [comment]
@Razzeee I added the list of checks that elm-analyse does and which ones I have implemented so far. Let me know or edit the list if there are some that you don't want to bring over. We can discuss, but I feel like there are some that we won't want to implement.

**Comment by Razzeee** [comment]
Agreed, I would however not try to do all in one PR, so that list might be best moved to an issue.

File: .github/workflows/npm-publish.yml [hunk 2]
```diff
@@ -1,47 +0,0 @@
-# This workflow will run tests using node and then publish a package to GitHub Packages when a release is created
```
Comment (Razzeee) [inline .github/workflows/npm-publish.yml:1]: Probably a rebase error

Comment (jmbockhorst) [reply .github/workflows/npm-publish.yml:1]: Then again, I'm not quite sure.

**Comment by jmbockhorst** [comment]
Ready for review. All of the tests from elm-analyse pass, but there could still be wrong cases out there. I left elm-analyse in place so that we can do some manual testing and compare the diagnostics. Once we feel it is good enough, we can disable these when elm-analyse is on, or just remove elm-analyse all together.

Of course we can now do much more than elm-analyse ever did, but as a start, I want to implement it a close to the existing behavior. 

Feel free to change/reword the diagnostics messages.

**Comment by Razzeee** [comment]
Great work, I will hopefully find some time next week to look at this, this weekend is pretty stuffed :)

File: src/providers/diagnostics/fileDiagnostics.ts [hunk 3]
```diff
@@ -6,6 +6,7 @@ export const enum DiagnosticKind {
   ElmAnalyse,
   ElmTest,
   TypeInference,
+  Elm,
```
Comment (Razzeee) [inline src/providers/diagnostics/fileDiagnostics.ts:9]: I'm conflicted on this name, might make more sense to name/report this as ElmLS? What do you think?

**Review by Razzeee — CHANGES_REQUESTED** [review]

File: src/providers/diagnostics/fileDiagnostics.ts (hunk 3 above)
Comment (jmbockhorst) [reply src/providers/diagnostics/fileDiagnostics.ts:9]: Yeah I didn't know what to name it. I'm fine with ElmLS.

File: src/providers/diagnostics/elmDiagnostics.ts [hunk 4]
```diff
@@ -0,0 +1,960 @@
...
+  }
+
+  public createDiagnostics = (
+    tree: Tree,
+    uri: string,
+    elmWorkspace: IElmWorkspace,
+  ): Diagnostic[] => {
+    try {
+      return [
+        ...this.getUnusedImportDiagnostics(tree),
```
Comment (Razzeee) [inline src/providers/diagnostics/elmDiagnostics.ts:41]: This is fine for now, but I think we will need at least the path ignore from here in the long run https://stil4m.github.io/elm-analyse/#/configuration

**Comment by Razzeee** [comment]
Those changes, a rebase and a global on/off switch and we're good to go.

**Comment by jmbockhorst** [comment]
Should be good now. Added a watch for the configuration so it updates dynamically.

File: test/diagnosticTests/elmDiagnostics.test.ts [hunk 5]
```diff
@@ -12,7 +12,7 @@ import { baseUri } from "../utils/mockElmWorkspace";
 import { getSourceFiles } from "../utils/sourceParser";
```
Comment (Razzeee) [inline test/diagnosticTests/elmDiagnostics.test.ts:12]: This file should be renamed

**Pull request merged**